report/web.py

```diff
@@ -89,6 +93,115 @@ class Target:
   fixer_prompt: Optional[str] = None
 
 
+class FileSystem:
```

Comment:
This already looks good to me, but some nits:
1. Instead of checking `if self._gcs_bucket is not None` in every function to know if this is the local filesystem or GCS bucket, would it be a good idea to use class inheritance and overwrite? E.g., `class LocalResultDir` and `class GCSResultDir(LocalResultDir)`.
2. How about hosting this class in its own file?

Reply:
I initially did try to have them be separate classes but to make that work cleanly, I would have to have a abstract base class, a GCS class, a local disk class, and a factory function that creates the right object based on the path. It just ended up being a lot of fluff code.

Reply:
> a abstract base class, a GCS class, a local disk class,

I reckon two classes should suffice for now: a base `class LocalResultDir` and an extended `class GCSResultDir(LocalResultDir)`.

> a factory function that creates the right object based on the path.

Would this be a workable solution for now?
```python
if path.startswith('gs://'):
  GCSResultDir(self._results_dir)
else:
  LocalResultDir(self._results_dir)
```

Again, the original implementation LGTM, no need to worry too much about this.

Reply:
Ah yeah I didn't consider using the local filesystem as a base class. I think both of these designs have their own drawbacks. Happy to chat about this next time we meet.

Same with a lot of these comments, I think we both have different styles that are both valid :)